src/lsm/lsm_cursor_bulk.c

```diff
@@ -90,17 +90,29 @@ __clsm_insert_bulk(WT_CURSOR *cursor)
 int
 __wt_clsm_open_bulk(WT_CURSOR_LSM *clsm, const char *cfg[])
 {
+	WT_CONFIG_ITEM cval;
 	WT_CURSOR *cursor, *bulk_cursor;
 	WT_LSM_TREE *lsm_tree;
 	WT_SESSION_IMPL *session;
+	int ordered;
 
 	bulk_cursor = NULL;
 	cursor = &clsm->iface;
 	lsm_tree = clsm->lsm_tree;
+	ordered = 1;	/* Default to ordered inserts */
 	session = (WT_SESSION_IMPL *)clsm->iface.session;
 
 	F_SET(clsm, WT_CLSM_BULK);
 
+	/*
+	 * Check for the undocumented unordered bulk flag, which is used when
+	 * doing index builds into LSM for existing trees.
+	 */
+	WT_RET(__wt_config_gets_def(session, cfg, "bulk", 0, &cval));
+	WT_ASSERT(session, cval.val != 0);
+	if (strncmp(cval.str, "unordered", strlen("unordered")) == 0)
+		ordered = 0;
+
```

Comment:
Sure - if it makes the index-create code more consistent, I'm happy to push
the functionality into btree.

On Wed, Aug 26, 2015 at 5:46 PM, Michael Cahill <notifications@github.com>
wrote:

> In src/lsm/lsm_cursor_bulk.c
> <https://github.com/wiredtiger/wiredtiger/pull/2150#discussion_r37954111>:
>
> >  	session = (WT_SESSION_IMPL *)clsm->iface.session;
> >
> >  	F_SET(clsm, WT_CLSM_BULK);
> >
> > +	/*
> > +	 * Check for the undocumented unordered bulk flag, which is used when
> > +	 * doing index builds into LSM for existing trees.
> > +	 */
> > +	WT_RET(__wt_config_gets_def(session, cfg, "bulk", 0, &cval));
> > +	WT_ASSERT(session, cval.val != 0);
> > +	if (strncmp(cval.str, "unordered", strlen("unordered")) == 0)
> > +		ordered = 0;
> > +
>
> Fair enough, I'm still trying to connect the dots to see what the
> __fill_index code will look like to take advantage of this feature: my
> hope was that there wouldn't have to be LSM-specific code in there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wiredtiger/wiredtiger/pull/2150/files#r37954111>.
>